python/ray/services.py

```diff
@@ -267,9 +267,19 @@ def start_ray_process(command,
     if os.environ.get(tmux_env_var) == "1":
         logger.info("Detected environment variable '%s'.", tmux_env_var)
         use_tmux = True
+    gdb_env_var = "RAY_{}_GDB".format(process_type.upper())
+    if os.environ.get(gdb_env_var) == "1":
+        logger.info("Detected environment variable '%s'.", gdb_env_var)
+        use_gdb = True
 
     if use_gdb:
-        raise NotImplementedError
+        # cwd is RAY_DIRECTORY/python/ray
+        cwd = os.path.abspath(os.path.dirname(__file__))
+        gdb_file = cwd + "/.gdbinit"
+        logger.info("Launch gdb with 'gdb {ray_file} -x {gdb_file}'".format(ray_file=command[-16], gdb_file=gdb_file))
+        run_args = ' '.join(["'{}'".format(arg) for arg in command[-15:]])
+        with open(gdb_file, "w") as gdbinit:
+            gdbinit.write("define run_ray\n\trun {}\nend".format(run_args))
```

Comment:
I'm a bit confused, we're writing to this file but never actually using it anywhere, right?

Reply:
we aren't using it in the commands but the file creates an alias in gdb that creates a command, `run_ray`, which has all the args needed to start the ray process in gdb. Then, using the command printed, the gdb instance sources the file and has the alias. 

Thus, the user would just have to copy the shell command printed to launch gdb in a new terminal instance and run `run_ray` to get start gdb running the ray process

Reply:
I see, the point is that users can specify breakpoints this way? I suppose that's a relevant use case, although I typically don't use gdb to debug segfaults without setting any breakpoints so just starting it automatically would be preferable.

I still think it'd be preferable to just automatically start it in gdb, though maybe we can support both use cases..

I'm pretty sure it's doable using https://stackoverflow.com/questions/14226563/how-to-read-and-execute-gdb-commands-from-a-file

Reply:
Ah I've been testing mainly with a `ipython` and was afraid of the two inputs clashing but realized that shouldn't be an issue. Pushed a commit that launches gdb automatically